Dockerfile.s3proxy

```diff
@@ -0,0 +1,40 @@
+FROM golang AS builder
+
+COPY . /rclone/
+WORKDIR /rclone/
+
+RUN \
+  CGO_ENABLED=0 \
+  make
+RUN ./rclone version
```

Comment:
every `RUN` creates an own layer and though increasing the size of the final container, its better to combine multiple runs into one
https://stackoverflow.com/questions/65072331/reducing-docker-container-size

Reply:
Builder stage won't be included in the final image. But I have tried to cut out the image size wherever possible